### PR TITLE
doc: horizontal-scaling docs

### PR DESCRIPTION
**Pull request by sigilioso**
Addresses #49.

We will probably need to extend this document when we document when scaling is needed (#48)

### CODE REVIEW

File: docs/Scaling.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+This kind of scaling comes without any complexity. It's as simple as updating the memory and/or the cpu for the corresponding machine where cluster the node is living.
+
+However, this approach may not be scalable for huge clusters, or simply we just don't want to have a single pod that consumes so many GBs of memory in our node. If so, you may need to take at <b>Horizontal Scaling<b/>.
+
+## Horizontal Scaling (Sharding)
+
+Horizontal Scaling is supported by setting up a configuration parameter which allows running several prometheus servers in agent mode to gather your data. This is known as Sharding.
+
+If you define `sharding.total_shards_count` value, the deployed StatefulSet will include as many replicas as you defined there. When it is used, the _configurator_ component will automatically include some additional relabel rules so each target will only be scraped by one prometheus server. Those
+rules rely on the target's address [hash-mod](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#relabel_config).
```
Comment (gsanchezgavier) [inline docs/Scaling.md:25]: i would maybe explain this with a bit more detail, like... each instance computes the this module of a hash and if the result matches the current instance the target is scraped. The selection is mutually exclusive meaning that a target will be only scraped by one shard as long as the shard count and the address of the target doesn't change.

Comment (gsanchezgavier) [reply docs/Scaling.md:25]: nit: perhaps a simple diagram could make the solution really easy to understand too

File: docs/Scaling.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+For example, if `custom-values.yaml` includes:
```
Comment (gsanchezgavier) [inline docs/Scaling.md:27]: maybe the diagram here along an example with some targets , perhaps just simplifying the example to 2 shards and having 2 targets (with different hashes mods)

File: docs/Scaling.md (hunk 1 above)
Comment (marcsanmi) [reply docs/Scaling.md]: Done in e946479fa0f27027615dd98d00fee9c9f1e6705a

File: docs/Scaling.md (hunk 2 above)
Comment (marcsanmi) [reply docs/Scaling.md:29]: Done in e946479fa0f27027615dd98d00fee9c9f1e6705a

**Review by gsanchezgavier — APPROVED** [review]
thanks

**Pull request merged**